gms/client/src/main/java/com/linkedin/dataset/client/Datasets.java

```diff
@@ -264,4 +273,55 @@ private DatasetKey toDatasetKey(@Nonnull DatasetUrn urn) {
     private DatasetUrn toDatasetUrn(@Nonnull DatasetKey key) {
         return new DatasetUrn(key.getPlatform(), key.getName(), key.getOrigin());
     }
+
+    /**
+     * Update an existing Dataset
+     */
+    public void update(@Nonnull final DatasetUrn urn, @Nonnull final Dataset dataset) throws RemoteInvocationException {
+        Request request = DATASET_ACTION_REQUEST_BUILDERS.createRequest(urn, toSnapshot(urn, dataset));
+        _client.sendRequest(request).getResponse();
+    }
+
+    // Copied from an unused method in Datasets resource.
+    static DatasetSnapshot toSnapshot(@Nonnull final DatasetUrn datasetUrn, @Nonnull final Dataset dataset) {
+        final List<DatasetAspect> aspects = new ArrayList<>();
+        if (dataset.getProperties() != null) {
+            aspects.add(ModelUtils.newAspectUnion(DatasetAspect.class, getDatasetPropertiesAspect(dataset)));
+        }
```

Comment:
where in this call do we associate the dataset's URN with the aspect data?

Reply:
On line 312 we create a "snapshot" model, which basically just has an URN and an array of associated aspects... so this is where that association happens :)